kv/rest_test.go

```diff
@@ -508,11 +510,15 @@ func getURL(url string, t *testing.T) string {
 }
 
 func httpDo(addr, method, path string, body io.Reader) (*http.Response, error) {
-	req, err := http.NewRequest(method, "http://"+addr+path, body)
+	req, err := http.NewRequest(method, "https://"+addr+path, body)
 	if err != nil {
 		return nil, err
 	}
-	return http.DefaultClient.Do(req)
+	return httpDoReq(req)
+}
+
+func httpDoReq(req *http.Request) (*http.Response, error) {
+	return client.CreateTestHTTPClient().Do(req)
```

Comment:
Loading certs from disk is relatively expensive. It doesn't make much difference for these simple tests but we really should change them to reuse an http.Client just to encourage good habits (or maybe just make a global var for a test client).